src/wallet/wallet.cpp

```diff
@@ -826,6 +833,29 @@ void CWallet::MarkDirty()
     }
 }
 
+bool CWallet::MarkReplaced(const uint256& originalHash, const uint256& newHash)
+{
+    LOCK(cs_wallet);
+
+    auto mi = mapWallet.find(originalHash);
+    if (mi == mapWallet.end())
+        return false;
+
+    CWalletTx& wtx = (*mi).second;
+
+    // Ensure for now that we're not overwriting data
+    assert(wtx.mapValue.count("replaced_by_txid") == 0);
+
+    wtx.mapValue["replaced_by_txid"] = newHash.ToString();
+
+    CWalletDB walletdb(strWalletFile, "r+");
+    walletdb.WriteTx(wtx);
+
+    NotifyTransactionChanged(this, originalHash, CT_UPDATED);
```

Comment:
Will look more, but I would want to find a specific case it is needed before adding MarkDirty here. I have another PR https://github.com/bitcoin/bitcoin/pull/9381 which eliminates MarkDirty from AddToWallet() and will get rid of this entire MarkReplaced() method, replacing it with an AddToWallet call.